### PR TITLE
Add Azure implementation of Security groups and Server groups

### PR DESCRIPTION
**Pull request by scotmoor**
Includes implementation of network and subnet support, bug fixes, and some cleanup

### CODE REVIEW

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/cache/AzureLoadBalancerCachingAgent.groovy [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import com.netflix.spinnaker.clouddriver.azure.resources.common.cache.Keys
+import com.netflix.spinnaker.clouddriver.azure.resources.common.cache.provider.AzureInfrastructureProvider
+import com.netflix.spinnaker.clouddriver.azure.resources.loadbalancer.model.AzureLoadBalancerDescription
+import com.netflix.spinnaker.clouddriver.azure.security.AzureCredentials
+import groovy.util.logging.Slf4j
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+
+@Slf4j
+class AzureLoadBalancerCachingAgent implements CachingAgent, AccountAware {
```
Comment (ajordens) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/cache/AzureLoadBalancerCachingAgent.groovy]: You guys will likely need to support OnDemand support ... to handle immediate updates when a forceCacheRefresh task happens.

Comment (milismsft) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/cache/AzureLoadBalancerCachingAgent.groovy]: Yes, I already start looking into adding that support given the latest round of PR changes affecting the other cloud providers.

**Comment by ajordens** [comment]
I didn't review in significant depth (3101 loc change makes that difficult) but I didn't see anything venturing outside of azure packages and artifacts so I suspect we're ok.

Will need a re-base.

**Comment by scotmoor** [comment]
re-base complete

**Comment by scotmoor** [comment]
PTAL - Anything else we need to do?

**Pull request merged**